### PR TITLE
feat: accept Undici File

### PR DESCRIPTION
**Pull request by juliusmarminge**
allows Undici File to be used for the utapi sdk functions

### CODE REVIEW

File: examples/with-serveractions/src/app/_actions.ts [hunk 1]
```diff
@@ -1,13 +1,18 @@
...
 /**
  * @see https://docs.uploadthing.com/api-reference/server#uploadfiles
  */
 export async function uploadFiles(fd: FormData) {
   const files = fd.getAll("files") as File[];
-  const uploadedFiles = await utapi.uploadFiles(files);
+  // const uploadedFiles = await utapi.uploadFiles(files);
+  const uploadedFiles = await utapi.uploadFiles([
+    new UndiciFile(["foo"], "foo.txt"),
+  ]);
```
Comment (juliusmarminge) [inline examples/with-serveractions/src/app/_actions.ts]: cc @p6l-richard this works for me

File: pnpm-lock.yaml [hunk 2]
```diff
@@ -174,8 +174,8 @@ importers:
         specifier: ^1.0.0
         version: 1.0.0(solid-js@1.7.11)
       undici:
-        specifier: 5.20.0
-        version: 5.20.0
+        specifier: ^5.20.0
+        version: 5.23.0
```
Comment (juliusmarminge) [inline pnpm-lock.yaml:178]: package managers do be kinda weird lol

File: packages/uploadthing/src/sdk/utils.ts [hunk 3]
```diff
@@ -89,7 +91,7 @@ export const uploadFilesInternal = async (
       Object.entries(fields).forEach(([key, value]) => {
         formData.append(key, value);
       });
-      formData.append("file", file);
+      formData.append("file", file as File);
```
[the remark this answers is not in the document]
Comment (juliusmarminge) [reply packages/uploadthing/src/sdk/utils.ts:94]: Not quite, Undici's File doesn't satisfy the built-in File type and thus not the FormData input either, but it works in runtime no problem, at least for all the tests I've made (https://github.com/pingdotgg/uploadthing/pull/325/commits/580681237c90157055235dfd5ba7783be9f9ded8)

[the remark this answers is not in the document]
Comment (markflorkowski) [reply packages/uploadthing/src/sdk/utils.ts:94]: Ah, I see. I misread what this coercion was doing. 👍

**Review by markflorkowski — APPROVED** [review]

**Pull request merged**